client/src/featureform/register.py

```diff
@@ -1590,7 +1591,89 @@ class Customer:
+        include_columns: List[str] = [],
+        exclude_columns: List[str] = [],
+        description: str = "",
+        schedule: str = "",
+        tags: List[str] = [],
+        properties: Dict[str, str] = {},
+    ):
+        """
+        Feature registration object.
+
+        **Example**
+        ```
+        @ff.entity
+        class Customer:
+        # Register mulitple columns from a dataframe as a feature
+            transaction_amount = ff.MultiFeature(
+                dataframe_name,
+                variant="quickstart",
+                inference_store=redis,
+                entity_column="CustomerID",
+                include_columns=["Amount", "Transaction Time"],
+            )
+        ```
 
+        Args:
+            inference_store (Union[str, OnlineProvider, FileStoreProvider]): Where to store for online serving.
+        """
+        pd_to_ff_datatype = {numpy.dtype('float64'): ScalarType.FLOAT64,
+        numpy.dtype('float32'): ScalarType.FLOAT32,
+        numpy.dtype('int64'): ScalarType.INT64,
+        numpy.dtype('int'): ScalarType.INT,
+        numpy.dtype('int32'): ScalarType.INT32,
+        numpy.dtype('O'): ScalarType.STRING,
+        numpy.dtype('str'): ScalarType.STRING,
+        numpy.dtype('bool'): ScalarType.BOOL}
+
+        register_columns = self.get_feature_columns(df, include_columns, exclude_columns, entity_column, timestamp_column)
+        register_features_list = []
+        for column_name in register_columns:
+            if timestamp_column != "":
+                feature = FeatureColumnResource(dataset[[entity_column, column_name, timestamp_column]], variant=variant, type=pd_to_ff_datatype[df[self.add_quotes(column_name)].dtype], inference_store=inference_store)
+            else:
+                feature = FeatureColumnResource(dataset[[entity_column, column_name]], variant=variant, type=pd_to_ff_datatype[df[self.add_quotes(column_name)].dtype], inference_store=inference_store)
+            feature.name = column_name
+            register_features_list.append(feature)
+            
+        self._resources = register_features_list
+
+    def get_feature_columns(self, df, include_columns, exclude_columns, entity_column, timestamp_column):
+        all_columns_set = set([self.strip_quotes(col) for col in df.columns])
+        include_columns_set = set(include_columns)
+        exclude_columns_set = set(exclude_columns)
+
+        if not include_columns_set.issubset(all_columns_set):
+            raise ValueError("Include columns must be in the dataframe")
+        if not exclude_columns_set.issubset(all_columns_set):
+            raise ValueError("Exclude columns must be in the dataframe")
+        if not include_columns_set.isdisjoint(exclude_columns_set):
+            raise ValueError("Include and exclude columns cannot have the same columns")
+        if len(include_columns_set) > 0:
+            return list(include_columns_set - {entity_column, timestamp_column})
```

Comment:
It might be better to add the `entity_column` and `timestamp_column` in the exclude set on line 1657